File: backend/app/schemas/conversation.py
import uuid
from datetime import datetime
from typing import List, Literal, Optional
from pydantic import BaseModel, ConfigDict, Field

from app.schemas.chat import ChatMessageRead


class ConversationListItem(BaseModel):
    """Summarized conversation session item for the owner's paginated inbox."""
    id: uuid.UUID
    created_at: datetime
    updated_at: datetime
    is_escalated: bool = False
    ticket_status: Optional[str] = None
    visitor_email: Optional[str] = None
    message_count: int = 0
    last_message_preview: Optional[str] = None
    last_message_role: Optional[str] = None

    model_config = ConfigDict(from_attributes=True)


class ConversationListResponse(BaseModel):
    """Paginated list envelope for owner's conversations inbox."""
    total: int
    limit: int
    offset: int
    items: List[ConversationListItem]


class ConversationDetailResponse(BaseModel):
    """Detailed view of a conversation including the complete chronological message transcript."""
    id: uuid.UUID
    organization_id: uuid.UUID
    created_at: datetime
    updated_at: datetime
    is_escalated: bool
    ticket_status: Optional[str] = None
    visitor_email: Optional[str] = None
    messages: List[ChatMessageRead]

    model_config = ConfigDict(from_attributes=True)


class TicketStatusUpdateRequest(BaseModel):
    """Request payload for updating the status of an escalated ticket."""
    status: Literal["open", "in_progress", "resolved"] = Field(
        ...,
        description="Target ticket lifecycle status (open, in_progress, resolved)",
    )


class TicketStatusUpdateResponse(BaseModel):
    """Response returned upon successful ticket status transition."""
    id: uuid.UUID
    ticket_status: str
    updated_at: datetime


class ConversationStatsResponse(BaseModel):
    """High-level conversation statistics and metrics for the organization dashboard."""
    total_conversations: int
    total_messages: int
    escalated_conversations: int
    active_last_24h: int

File: frontend/types/conversation.ts
export type TicketStatus = "open" | "in_progress" | "resolved";

export type MessageRole = "visitor" | "assistant" | "system";

export interface CitationItem {
  document_id: string;
  title: string;
}

export interface ChatMessage {
  id: string;
  role: MessageRole;
  content: string;
  created_at: string;
  citations?: CitationItem[] | null;
  metadata?: Record<string, any> | null;
}

export interface ConversationSummary {
  id: string;
  created_at: string;
  updated_at: string;
  is_escalated: boolean;
  ticket_status?: TicketStatus | null;
  visitor_email?: string | null;
  message_count: number;
  last_message_preview?: string | null;
  last_message_role?: string | null;
  is_new?: boolean;
}

export interface ConversationDetail {
  id: string;
  organization_id: string;
  created_at: string;
  updated_at: string;
  is_escalated: boolean;
  ticket_status?: TicketStatus | null;
  visitor_email?: string | null;
  messages: ChatMessage[];
}

export interface ConversationListResponse {
  total: number;
  limit: number;
  offset: number;
  items: ConversationSummary[];
}

export interface ConversationStats {
  total_conversations: number;
  total_messages: number;
  escalated_conversations: number;
  active_last_24h: number;
}

export type InboxFilterTab = "all" | "escalated";

export interface InboxViewState {
  selectedConversationId: string | null;
  activeFilter: InboxFilterTab;
  searchQuery: string;
  page: number;
  pageSize: number;
  isMobileDetailOpen: boolean;
}
